src/main/kotlin/Todo/ToDoList.kt

```diff
@@ -0,0 +1,34 @@
+package Todo
+
+import Utilities.FileInterface
+import Utilities.JsonGenerator
+import java.io.FileNotFoundException
+
+class ToDoList(private val filePath: String, private val fileIo: FileInterface, private val jsonGenerator: JsonGenerator) {
+
+    fun addTask(task: Task): String {
+        val taskId = calculateId()
+        task.setId(taskId)
+        val jsonTask = jsonGenerator.resourceToJson(task)
+        fileIo.writeResource(filePath, jsonTask)
+        return task.body
+    }
+
+    fun retrieveList(): List<String> {
+        return try {
+            val tasks = fileIo.readResource(filePath)
+            tasks.split("\n").toList()
+        }catch (e: FileNotFoundException) {
+            listOf()
+        }
+    }
+
+    private fun calculateId(): Int {
+        val listOfTasks = retrieveList()
+        return if (listOfTasks.isEmpty()) {
+            1
+        } else {
+            listOfTasks.size
+        }
+    }
```

Comment:
We don't have delete tasks now, but it's something we might add. Would that cause some issues? Is that something you're going to handle later?

Also, what requires you to have an ID?

Reply:
The motivation behind the decision to include an ID was to mimic how data Is stored in a relational database. As for deleting tasks, this function will have to be changed since IDs should be unique and not reused. Instead of utilizing the size of the list, I can pluck the last `Task` and increment its `id` value by 1 to calculate the new `id` value for the new `Task`. However, using this approach if the last item in the list were to be deleted, its `id` value would be reused.